test/validation/api/classification/odp_classification_common.c

```diff
@@ -359,6 +361,14 @@ odp_packet_t create_packet(cls_packet_info_t pkt_info)
 		next_hdr = ODPH_IPPROTO_IGMP;
 		l4_hdr_len = ODP_IGMP_HLEN;
 		break;
+	case CLS_PKT_L4_AH:
+		next_hdr = ODPH_IPPROTO_AH;
+		l4_hdr_len = ODPH_AHHDR_LEN;
```

Comment:
Now I noticed that the l4_hdr_len here does not match the ah_len field below. l4_hdr_len here is 12 but ah_len indicates 24-byte AH header. So the ODP under test might think that the AH packet is truncated before it processes the classification rules.

There is a similar problem with ESP in that l4_hdr_len does not include the trailer (plus padding and/or actual payload) that must be in the packet, so an ODP implementation might think the ESP packet is invalid (as no real ESP packet can be that short) and stop processing it further. The l4_hdr_len here (or resulting packet size that gets calculated) should be 4 bytes longer than now.

This is getting slightly messy and might also be unnecessary, but I guess it would be good to have the test packets to look like valid packets.